### PR TITLE
Use ceres::Manifolds instead of ceres::LocalParameterization.

### PR DESCRIPTION
**Pull request by vrabaud**
Use ceres::Manifolds instead of ceres::LocalParameterization.
The latter is deprecated.

This fixes https://github.com/opencv/opencv_contrib/issues/3218

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch

### CODE REVIEW

**Review by victor1234 — CHANGES_REQUESTED** [review]

File: modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc [hunk 1]
```diff
@@ -494,8 +499,13 @@ void EuclideanBundleCommonIntrinsics(
       constant_translation.push_back(4);
       constant_translation.push_back(5);
 
+#if CERES_VERSION_MAJOR >= 2 && CERES_VERSION_MINOR >= 1
+      constant_translation_parameterization =
```
Comment (victor1234) [inline modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:503]: You mean `constant_translation_manifold`?

Comment (vrabaud) [reply modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:503]: indeed, thx. Fixed

**Review by victor1234 — APPROVED** [review]

File: modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc [hunk 2]
```diff
@@ -485,7 +486,11 @@ void EuclideanBundleCommonIntrinsics(
     PackCamerasRotationAndTranslation(tracks, *reconstruction);
 
   // Parameterization used to restrict camera motion for modal solvers.
+#if CERES_VERSION_MAJOR >= 2 && CERES_VERSION_MINOR >= 1
```
Comment (alalek) [inline modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:489]: This check is not accurate as 3.0 version returns `false`.

File: modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc [hunk 3]
```diff
@@ -538,8 +548,13 @@ void EuclideanBundleCommonIntrinsics(
       }
 
       if (bundle_constraints & BUNDLE_NO_TRANSLATION) {
+#if CERES_VERSION_MAJOR >= 2 && CERES_VERSION_MINOR >= 1
```
Comment (asmorkalov) [inline modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:551]: The same issue with 3.0.

File: modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc [hunk 4]
```diff
@@ -586,10 +601,17 @@ void EuclideanBundleCommonIntrinsics(
     // Always set K3 constant, it's not used at the moment.
     constant_intrinsics.push_back(OFFSET_K3);
 
+#if CERES_VERSION_MAJOR >= 2 && CERES_VERSION_MINOR >= 1
```
Comment (asmorkalov) [inline modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc:604]: And the same.

File: modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc (hunk 3 above)
Comment (vrabaud) [reply modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc]: Done, sorry about that.

File: modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc (hunk 4 above)
Comment (vrabaud) [reply modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc]: Done

**Review by asmorkalov — APPROVED** [review]
Thanks!

**Comment by Chuvi-w** [comment]
https://github.com/opencv/opencv_contrib/blob/aca2aec7808ae30708f5d25dcf2edb098655a56a/modules/sfm/src/libmv_light/libmv/simple_pipeline/bundle.cc#L552

error: cannot initialize a parameter of type 'ceres::LocalParameterization *' with an lvalue of type 'ceres::SubsetManifold *'

**Comment by alalek** [comment]
@Chuvi-w Fixup is here: #3274

**Pull request merged**